internal/files/linkedfiles.go

```diff
@@ -23,43 +22,168 @@ import (
 
 const linkExtension = ".link"
 
+// PackageLinks represents linked files grouped by package.
+type PackageLinks struct {
+	PackageName string
+	Links       []string
+}
+
+// CheckLinkedFiles checks if all linked files in the given directory are up-to-date.
+// Returns a list of outdated links that need updating.
+func CheckLinkedFiles(fromDir string) ([]Link, error) {
+	repoRoot, err := FindRepositoryRootDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("finding repository root: %w", err)
+	}
+
+	root, err := os.OpenRoot(repoRoot)
+	if err != nil {
+		return nil, fmt.Errorf("opening repository root: %w", err)
+	}
+
+	return AreLinkedFilesUpToDate(root, fromDir)
+}
+
+// UpdateLinkedFiles updates the checksums of all outdated linked files in the given directory.
+// Returns a list of links that were updated.
+func UpdateLinkedFiles(fromDir string) ([]Link, error) {
+	repoRoot, err := FindRepositoryRootDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("finding repository root: %w", err)
+	}
+
+	root, err := os.OpenRoot(repoRoot)
+	if err != nil {
+		return nil, fmt.Errorf("opening repository root: %w", err)
+	}
+
+	return UpdateLinkedFilesChecksums(root, fromDir)
+}
+
+// IncludeLinkedFilesFromPath copies all linked files from the source directory to the target directory.
+// This is used during package building to include linked files in the build output.
+func IncludeLinkedFilesFromPath(fromDir, toDir string) ([]Link, error) {
+	repoRoot, err := FindRepositoryRootDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("finding repository root: %w", err)
+	}
+
+	root, err := os.OpenRoot(repoRoot)
+	if err != nil {
+		return nil, fmt.Errorf("opening repository root: %w", err)
+	}
+
+	return IncludeLinkedFiles(root, fromDir, toDir)
+}
+
+// ListLinkedFilesByPackage returns a mapping of packages to their linked files that reference
+// files from the given directory.
+func ListLinkedFilesByPackage(fromDir string) ([]PackageLinks, error) {
+	repoRoot, err := FindRepositoryRootDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("finding repository root: %w", err)
+	}
+
+	root, err := os.OpenRoot(repoRoot)
+	if err != nil {
+		return nil, fmt.Errorf("opening repository root: %w", err)
+	}
+
+	return LinkedFilesByPackageFrom(root, fromDir)
+}
+
```

Comment:
These methods open the root, and then call another method that uses this root and actually provides this functionality. This duplication of methods may lead in the future to create confusion on what method to use. Should I use `ListLinkedFilesByPackage`, or `LinkedFilesByPackageFrom`? The first is safer, but the second might be used by mistake.

We also have the `CreateLinksFSFromPath` that also opens the root and creates an object that wraps this root and provides some operations on it.

Couldn't these methods also be part of the `LinkFS` object?

Then for any operation we would use first `CreateLinksFSFromPath`, and then call `linkFS.ListLinkedFilesByPackage` for example.